theatre/studio/src/panels/SequenceEditorPanel/DopeSheet/Right/BasicKeyframedTrack/KeyframeEditor/useTempTransactionEditingTools.tsx

```diff
@@ -0,0 +1,51 @@
+import getStudio from '@theatre/studio/getStudio'
+import type {SerializableValue} from '@theatre/shared/utils/types'
+import type {
+  CommitOrDiscard,
+  ITransactionPrivateApi,
+} from '@theatre/studio/StudioStore/StudioStore'
+import type {IEditingTools} from '@theatre/studio/propEditors/utils/IEditingTools'
+import {useMemo} from 'react'
+
+/**
+ * This function takes a function `writeTx` that sets a value in the private Studio API and
+ * returns a memoized editingTools object which contains three functions:
+ * - `temporarilySetValue` - uses `writeTx` to set a value that can be discarded
+ * - `discardTemporaryValue` - if `temporarilySetValue` was called, discards the value it set
+ * - `permanentlySetValue` - uses `writeTx` to set a value
+ *
+ * @param writeTx - a function that uses a value to perform an action using the
+ * private Studio API.
+ * @returns an editingTools object that can be passed to `DeterminePropEditorForKeyframe` or
+ * `DetailDeterminePropEditor` and is used by the prop editors in `simplePropEditorByPropType`.
+ */
+export function useTempTransactionEditingTools<T extends SerializableValue>(
```

Comment:
Since this just wraps `createTempTransactionEditingTools()` in a `useMemo()` with no other react-specific logic, wouldn't it be better if use-sites used `createTempTransactionEditingTools()` directly (and wrap it in `useMemo()` themselves?).

Reply:
Heard. It seems like a good idea, but also, I think there is a benefit to placing the `deps` list next to the implementation. So, the callers don't have to be aware of how it works internally to make a decision about when to re-compute.